### PR TITLE
feat(1225): implement delete activity trace associations

### LINKED ISSUE #1225: [FE] Créer le composant AssociateElementsSelector
### Description

- Créer le composant `AssociateElementsSelector`
- Il prend en props une liste d'éléments avec a minima id, title et type (trace / compétence)
- Il utilise `SelectorOverlay` et  `AssociateElementCard`
- Il a un `defineModel` avec les id des éléments sélectionnés

### Éléments UI de travail

[lien Figma](https://www.figma.com/design/fA1dOIZs4Q3wTIJLSNQ5qk/COFOLIO?node-id=40019225-204596&t=Kp09jLRR2RukYVLw-4)

Image

### PR DESCRIPTION
**Pull request by p-beger**
### Brief description of changes applied
This PR implements the deletion of activity trace associations. It introduces the ability to remove associations directly from the activity details, improving user control and data management.

**Main changes:**
- ✨ Implements delete functionality for activity trace associations
- ✅ Adds and updates unit tests for deletion logic
- 🎨 Refactors modal and selector components for better maintainability
- 🔥 Cleans up related stubs and test files

---

### Reference to an Issue, Feature, Task, User Story or another PR
Closes https://github.com/avenirs-esr/AVENIRS-Project/issues/1225

---

### Documentation
- Unit tests added and updated for delete associations feature
- Refactored modal and selector components for clarity
- Updated stubs for new logic

**Modified files:**
- `src/features/student/buildProject/views/ProjectActivityDetailedView/components/DeleteActivityAssociatedTracesModal.stub.ts`
- `src/features/student/buildProject/views/ProjectActivityDetailedView/components/DeleteActivityAssociatedTracesModal.test.ts`
- `src/features/student/buildProject/views/ProjectActivityDetailedView/components/DeleteActivityAssociatedTracesModal.vue`
- `src/features/student/buildProject/views/ProjectActivityDetailedView/components/tabs/AssociatedElementsTab/AssociatedElementsTab.test.ts`
- `src/features/student/buildProject/views/ProjectActivityDetailedView/components/tabs/AssociatedElementsTab.vue`
- `src/features/student/buildProject/views/ProjectActivityDetailedView/components/CompactCardSelector/CompactCardSelector.vue`
- `src/features/student/buildProject/views/ProjectActivityDetailedView/components/SelectorOverlay/SelectorOverlay.vue`
- `src/features/student/buildProject/views/ProjectActivityDetailedView/components/DeleteAssociationsConfirmModal.test.ts`
- `src/features/student/buildProject/views/ProjectActivityDetailedView/components/DeleteAssociationsConfirmModal.vue`
- `src/features/student/buildProject/views/ProjectActivityDetailedView/components/DeleteAssociationsModal.test.ts`
- `src/features/student/buildProject/views/ProjectActivityDetailedView/components/DeleteAssociationsModal.vue`

---

### Target Branch
`develop`

---

### Additional Notes
This PR enhances the user experience by allowing direct management of activity trace associations. The refactor also improves code maintainability and test coverage.

**Technical decisions:**
- Centralized deletion logic in modal components
- Improved test coverage for edge cases
- Cleaned up unused stubs and test exports

---

### Known Limitations or Side Effects
No known limitations or side effects.

---

## ✅ Checklist

- [ ] a11y tested (if the PR includes frontend changes)
- [x] Tests provided (including unit and integration tests for both frontend and backend)
- [x] i18n handled (texts are translated)
- [ ] Performance tests
- [x] No unnecessary code (e.g., debug logs, commented code)
- [x] Code style and formatting rules respected (linting, conventions, etc.)
- [ ] Semantic Versioning respected
- [ ] Add to `CHANGELOG.md` file all properties and database change and details for the update process

### CODE REVIEW

File: src/features/student/buildProject/views/ProjectActivityDetailedView/components/overlays/modals/DeleteActivityAssociatedTracesModal/DeleteActivityAssociatedTracesModal.vue [hunk 1]
```diff
@@ -2,16 +2,22 @@
...
+import CompactCardSelector from '@/features/student/global/components/cards/CompactCardSelector/CompactCardSelector.vue'
 import DeleteAssociationsModal from '@/features/student/global/components/overlays/modals/DeleteAssociationsModal/DeleteAssociationsModal.vue'
+import { ICONS } from '@/features/student/global/icons'
 import { useToasterStore } from '@/store'
 import { useI18n } from 'vue-i18n'
 
 export interface DeleteActivityAssociatedTracesModalProps {
   show: boolean
   declaredActivityId: string
+  associations: {
```
Comment (elrhourha) [inline src/features/student/buildProject/views/ProjectActivityDetailedView/components/overlays/modals/DeleteActivityAssociatedTracesModal/DeleteActivityAssociatedTracesModal.vue]: @p-beger je pense qu'il faut créer un type global pour l'id et label et l'utiliser partout, on l'a partout maintenant

Comment (p-beger) [reply src/features/student/buildProject/views/ProjectActivityDetailedView/components/overlays/modals/DeleteActivityAssociatedTracesModal/DeleteActivityAssociatedTracesModal.vue]: C'est fait @elrhourha :+1:

**Review by elrhourha — APPROVED** [review]
